Compare-NET-Objects/IgnoreOrderTypes/IgnoreOrderLogic.cs

```diff
@@ -236,6 +234,15 @@ private string GetMatchIndex(ComparisonResult result, List<string> spec, object
                         $"Invalid CollectionMatchingSpec.  No such property {item} for type {currentObject.GetType().Name} ");
                 }
 
+                // Ensure that we will not compare indexer property.
+                var indexParameters = info.GetIndexParameters();

```

Comment:
I added explicit check on indexers to replace reflection error with our own exception.